File: base/src/main/java/io/quarkus/code/config/SegmentConfig.java
package io.quarkus.code.config;

import io.smallrye.config.ConfigMapping;
import java.util.Optional;
import java.util.OptionalInt;

@ConfigMapping(prefix = "io.quarkus.code.segment")
public interface SegmentConfig {
    Optional<String> writeKey();

    OptionalInt flushQueueSize();

    OptionalInt flushIntervalSeconds();

    default String writeKeyForDisplay() {
        return writeKey().filter(s -> !s.isBlank()).orElse("UNDEFINED");
    }
}

File: api/src/main/kotlin/io/quarkus/code/config/SegmentConfig.kt
package io.quarkus.code.config

import io.smallrye.config.ConfigMapping
import io.smallrye.config.WithName
import java.util.*

@ConfigMapping(prefix = "io.quarkus.code.segment")
interface SegmentConfig {
    @get:WithName("write-key")
    val writeKey: Optional<String>

    @get:WithName("flushQueueSize")
    val flushQueueSize: OptionalInt

    @get:WithName("flushIntervalSeconds")
    val flushIntervalSeconds: OptionalInt

    fun writeKeyForDisplay(): String {
        return writeKey.filter(String::isNotBlank).map { it }.orElse("UNDEFINED")
    }
}